zipkin-ui/js/page/trace.js

```diff
@@ -17,19 +18,27 @@ const TracePageComponent = component(function TracePage() {
     TraceData.attachTo(document, {
       traceId: this.attr.traceId
     });
-    this.on(document, 'tracePageModelView', function(ev, modelview) {
-      this.$node.html(traceTemplate(modelview));
+    this.on(document, 'tracePageModelView', function(ev, data) {
+      this.$node.html(traceTemplate(data.modelview));
 
       FilterAllServicesUI.attachTo('#filterAllServices', {
         totalServices: $('.trace-details.services span').length
       });
       FullPageSpinnerUI.attachTo('#fullPageSpinner');
+      JsonPanelUI.attachTo('#jsonPanel');
       ServiceFilterSearchUI.attachTo('#serviceFilterSearch');
       SpanPanelUI.attachTo('#spanPanel');
       TraceUI.attachTo('#trace-container');
       FilterLabelUI.attachTo('.service-filter-label');
       ZoomOut.attachTo('#zoomOutSpans');
 
+      this.$node.find('#traceJsonLink').click(e => {
```

Comment:
@eirslett you're right, this ends up being cleaner anyways, it never really belonged in trace.js